src/app/games/trexo/TrexoRules.ts

```diff
@@ -0,0 +1,158 @@
+                    if (MGPNode.getScoreStatus(squareScore) === SCORE.VICTORY) {
+                        if (pieceOwner === lastPlayer) {
+                            victoryOfLastPlayer.push(coord);
+                        } else {
+                            victoryOfNextPlayer.push(coord);
+                        }
+                    }
+                }
+            }
+        }
+        if (victoryOfNextPlayer.length > 0) {
+            return victoryOfNextPlayer;
+        } else {
+            return victoryOfLastPlayer;
+        }
+    }
+    public getGameStatus(node: TrexoNode): GameStatus {
+        const state: TrexoState = node.gameState;
+        const lastPlayer: Player = state.getCurrentOpponent();
+        let lastPlayerAligned5: boolean = false;
+        for (let x: number = 0; x < TrexoState.SIZE; x++) {
+            // for every column, starting from the bottom of each column
+            for (let y: number = 0; y < TrexoState.SIZE; y++) {
+                // while we haven't reached the top or an empty space
+                const pieceOwner: PlayerOrNone = state.getPieceAtXY(x, y).getOwner();
+                if (pieceOwner.isPlayer()) {
+                    const squareScore: number = TrexoRules.getSquareScore(state, new Coord(x, y));
+                    if (MGPNode.getScoreStatus(squareScore) === SCORE.VICTORY) {
+                        if (pieceOwner === lastPlayer) {
+                            lastPlayerAligned5 = true;
+                        } else {
+                            return GameStatus.getDefeat(lastPlayer);
+                        }
+                    }
+                }
+            }
+        }
+        if (lastPlayerAligned5) {
+            return GameStatus.getVictory(lastPlayer);
+        }
+        return GameStatus.ONGOING;
+    }
+    public getLegalMoves(state: TrexoState): TrexoMove[] {
+        const moves: TrexoMove[] = [];
+        for (let x: number = 0; x < TrexoState.SIZE; x++) {
+            for (let y: number = 0; y < TrexoState.SIZE; y++) {
+                const upOrleftCord: Coord = new Coord(x, y);
+                if (x + 1 < TrexoState.SIZE) {
+                    const rightCoord: Coord = new Coord(x + 1, y);
+                    moves.push(...this.getPossiblesMoves(state, upOrleftCord, rightCoord));
+                }
+                if (y + 1 < TrexoState.SIZE) {
+                    const downCoord: Coord = new Coord(x, y + 1);
+                    moves.push(...this.getPossiblesMoves(state, upOrleftCord, downCoord));
+                }
+            }
+        }
+        return moves;
+    }
+    public getPossiblesMoves(state: TrexoState, first: Coord, second: Coord): TrexoMove[] {
+        const firstStack: TrexoPieceStack = state.getPieceAt(first);
+        const secondStack: TrexoPieceStack = state.getPieceAt(second);
+        let tileHideEntirelyOneTile: boolean;
+        if (firstStack.getUpperTileId() === -1) {
```

Comment:
définir `isGround()` comme `this.getUpperTileId() === -1` pour clarifier ?
Aussi : tileHide*s*EntirelyOneTile